packages/jest-resolve/src/isBuiltinModule.ts

```diff
@@ -23,5 +23,10 @@ const BUILTIN_MODULES = new Set(
 );
 
 export default function isBuiltinModule(module: string): boolean {
-  return BUILTIN_MODULES.has(module);
+  // https://nodejs.org/api/modules.html#modules_core_modules
+  // Core modules can also be identified using the node: prefix, for instance, require('node:http')
+
+  return BUILTIN_MODULES.has(
+    module.startsWith('node:') ? module.slice(5) : module,
```

Comment:
Here we can avoid the ternary operator, and do it with Regex:
```js
export default function isBuiltinModule(module: string): boolean {
  moduleName = moduleName.replace(/^node:/, '');
  return BUILTIN_MODULES.has(module);
}
```

See discussion: https://github.com/facebook/jest/pull/11638#discussion_r663490803

Reply:
Hey @Divlo,

1) Just thought, that regex might be slower.
2) Argument mutation is not "the best practice".

```js
var Benchmark = require('benchmark')
var suite = new Benchmark.Suite;

const moduleName = 'node:foobar'
suite
  .add('String#replace', function() {
    moduleName.replace(/^node:/, '');
  })
  .add('String#startsWith', function() {
    moduleName.startsWith('node:') ? moduleName.slice(5) : moduleName;
  })
  .on('cycle', function(event) {
    console.log(String(event.target));
  })
  .on('complete', function() {
    console.log('Fastest is ' + this.filter('fastest').map('name'));
  })
  .run({ 'async': true });
```
```
 node bench.js
String#replace x 17,913,326 ops/sec ±1.19% (87 runs sampled)
String#startsWith x 57,254,145 ops/sec ±0.73% (88 runs sampled)
```

Reply:
Indeed! :+1: 
I also ran it on my laptop :
```txt
String#replace x 25,474,843 ops/sec ±0.37% (96 runs sampled)
String#startsWith x 86,410,568 ops/sec ±0.21% (98 runs sampled)
Fastest is String#startsWith
```

Great one, will also change this in my PR.